cpp/src/copying/contiguous_split.cu

```diff
@@ -430,7 +416,7 @@ contiguous_split_result alloc_and_copy(cudf::table_view const& t,
 
```

Comment:
After writing the message below, I realized that this isn't a `for_each` - we are calculating `total_size` - this is an `accumulate` - or even better seeing as this is a `.cu` file, `thrust::transform_reduce` with a `thrust::counting_iterator`. Code should be something like:
```cpp
  auto total_size = std::transform_reduce(
    t.begin(), 
    t.end(), 
    thrust::make_counting_iterator(0),
    size_t{0},
    thrust::plus{},
    [&split_info](cudf::column_view const& c, size_type column_index) {
      return cudf::experimental::type_dispatcher(
        c.type(), column_buffer_size_functor{}, c, split_info[column_index]);
    });
```

==== 
I realize you didn't modify the code above but I am interested in people's thought on this. If I need a modifiable value in a lambda passed to an algorithm, I typically mark the lamba as `mutable` and just init the variable by value instead of passing by reference so instead of this:
```cpp
  size_t total_size      = 0;
  size_type column_index = 0;
  std::for_each(
    t.begin(), t.end(), [&total_size, &column_index, &split_info](cudf::column_view const& c) {
      total_size += cudf::experimental::type_dispatcher(
        c.type(), column_buffer_size_functor{}, c, split_info[column_index]);
      column_index++;
    });
```
you have something like this:
```cpp
  std::for_each(
    t.begin(), t.end(), [total_size = 0, column_index = 0, &split_info](cudf::column_view const& c) mutable {
      total_size += cudf::experimental::type_dispatcher(
        c.type(), column_buffer_size_functor{}, c, split_info[column_index]);
      column_index++;
    });
```